tests/integration/IntegrationTest.js

```diff
@@ -0,0 +1,107 @@
+'use strict'
+
+const tape = require('tape')
+const http = require('http')
+const supertest = require('supertest')
+const {createTestEndpoint, removeTestEndpoint} = require('../testUtils')
+
+// Create an endpoint for the tests
+createTestEndpoint()
+
+const app = require('../../src/index')
+
+// Remove the test endpoint folder
+removeTestEndpoint()
+
+tape.test('Parsing Integration Tests', t => {
+  tape.onFinish(() => {
+    server.close()
+  })
+
+  const server = http.createServer(app.callback())
+  const request = supertest(server)
+
+  t.test(
+    'should fail to map when parsing fails (body format and content-type mismatch)',
+    t => {
+      const payload = '{length: 8.0, width: 4.0}'
+
+      request
+        .post('/integrationTest')
+        .send(payload)
+        .expect(400)
+        .set('Content-Type', 'application/xml')
+        .set('Accept', 'application/json')
+        .end((err, res) => {
+          if (err) {
+            t.fail(err)
+          }
+
+          t.equals(
+            res.text.match(/Parsing incoming body failed: Bad Request/).length,
+            1
+          )
+          t.end()
+        })
+    }
+  )
+
+  t.test(
+    'should fail to map when validation fails (required property missing)',
+    t => {
+      const payload = `<xml>
+        <surname>Li</surname>
+        <email>jet@openhim.org</email>
+        </xml>`
+
+      request
+        .post('/integrationTest')
+        .send(payload)
+        .expect(400)
+        .set('Content-Type', 'application/xml')
+        .set('Accept', 'application/json')
+        .end((err, res) => {
+          if (err) {
+            t.fail(err)
+          }
+
+          t.equals(res.text.match(/Validation failed:/).length, 1)
```

Comment:
Instead of matching the string value in the response text, could we not just verify the entire response text?

Reply:
fixed in https://github.com/jembi/openhim-mediator-mapping/pull/27/commits/0c2bb62586146464f86afc15d7b519f8845737b8